src/components/modal/Modal.svelte

```diff
@@ -0,0 +1,408 @@
+      document.body.removeEventListener("click", setWrapClickClose);
+      document.removeEventListener("click", setClickPosition);
+    }
+  });
+
+  $: modalClasses = computeModalClasses();
+
+  function computeModalClasses() {
+    let classes = "ant-modal";
+    if (modalType !== "standard") {
+      classes += ` ant-modal-confirm ant-modal-confirm-${modalType}`;
+    }
+    return classes;
+  }
+
+  $: if (!width) {
+    width = modalType === "standard" ? "520px" : "416px";
+    modalStyleString = `width: ${width};`;
+  }
+
+  $: if (visible) {
+    setTimeout(() => {
+      scrollableBody = false;
+      localVisible = true;
+    }, 50);
+  } else {
+    destroy();
+  }
+
+  $: if (Object.keys(verticalPosition)) {
+    for (const [key, value] of Object.entries(verticalPosition)) {
+      if (["top", "bottom"].includes(key))
+        modalStyleString += ` ${key}: ${value};`;
+    }
+  }
+
+  $: if (!icon) {
+    switch (modalType) {
+      case "info":
+        icon = InfoCircleOutlined;
+        break;
+      case "success":
+        icon = CheckCircleOutlined;
+        break;
+      case "error":
+        icon = CloseCircleOutlined;
+        break;
+      case "warning":
+        icon = ExclamationCircleOutlined;
+        break;
+      default:
+        icon = null;
+    }
+  }
+
+  // We want to close the modal if we clicked outside of it
+  $: if (localVisible && mask && maskClosable) {
+    document.body.addEventListener("click", setWrapClickClose);
+  }
+
+  // Set click position each time there is a click
+  if (typeof window !== "undefined" && window.document) {
+    window.document.addEventListener("click", setClickPosition);
+  }
+
+  //*** Transition logic start ***//
+  $: if (clickPosition) {
+    fadescale = function(node, { duration }) {
+      let transformOrigin = "";
+      const dialogNode = document.querySelector(".ant-modal-content");
+      const elOffset = offset(dialogNode);
+      transformOrigin = ` transform-origin: ${clickPosition.x -
+        elOffset.left}px ${clickPosition.y - elOffset.top}px;`;
+
+      return {
+        duration,
+        css: t => `opacity: ${t}; transform: scale(${t});${transformOrigin}`
+      };
+    };
+  }
+
+  function getScroll(w, top) {
+    let ret = w[`page${top ? "Y" : "X"}Offset`];
+    const method = `scroll${top ? "Top" : "Left"}`;
+    if (typeof ret !== "number") {
+      const d = w.document;
+      ret = d.documentElement[method];
+      if (typeof ret !== "number") {
+        ret = d.body[method];
+      }
+    }
+    return ret;
+  }
+
+  function offset(el) {
+    const rect = el.getBoundingClientRect();
+    const pos = {
+      left: rect.left,
+      top: rect.top
+    };
+    const doc = el.ownerDocument;
+    const w = doc.defaultView || doc.parentWindow;
+    pos.left += getScroll(w);
+    pos.top += getScroll(w, true);
+    return pos;
+  }
+  //*** Transition logic end ***//
+
+  async function onOkLocal() {
```

Comment:
why do we expect onOk and onCancel to be promises ?

Reply:
I thought it would be good to give that option. If we don't and they are promises then the modal will close before they are completed. Do you think that's the wrong approach?

Reply:
For now I removed this and it's much simpler. I think we can add this in the future if there is a need, but it seems like the function should not be a promise.